sql/sql_statistics.h

```diff
@@ -153,6 +153,24 @@ class Histogram_base : public Sql_alloc
   virtual void serialize(Field *to_field)= 0;
 
   virtual Histogram_type get_type()=0;
+
+  virtual uint get_width()=0;
+
+  virtual void init_for_collection(MEM_ROOT *mem_root, Histogram_type htype_arg, ulonglong size)=0;
+
+  virtual bool is_available()=0;
```

Comment:
Need to look at histogram::is_available(). 

This function made sense in the old code, where it was possible that Histogram object is present but is "empty".  This should no longer be the case. (I am not sure whether this belongs to this milestone.. just making a note about something to think about)